### ISSUE TITLE
[SL4A] Touchpad Freezes Randomly

### ISSUE THREAD

**Issue by justrogerz** [linux-surface/linux-surface#1085]
Hi, I switched to Arch with linux-surface some days ago, it's all almost perfect except for the touchpad, which just randomly stops moving and then comes back to life after a couple seconds... It's not a big deal but it can be really annoying sometimes.This also happens with the keyboard, but very less frequently. It never happened on Windows btw.

### Environment

- Hardware model: Surface Laptop 4 (AMD)
- Kernel version: Linux surface 6.2.0-arch1-1-surface
- Distribution: Arch

<details><summary>`dmesg` output</summary>

```
it was too long i couldn't copy it all

```

</details>

**Comment by qzed** [comment]
Maybe related to https://github.com/linux-surface/linux-surface/issues/1026?

**Comment by justrogerz** [comment]
> Maybe related to #1026?

I don't think so, because in #1026

**Comment by justrogerz** [comment]
> Maybe related to #1026?

I don't think so, because in #1026 the issue happens because of the pressure level, which is only present in Surface Studio... in Surface Laptop like mine there's no pressure sensor; I noticed that the freezes happen sometimes after doing some 2-finger or 3-finger gestures, but not always. I think it's a driver related issue.

**Comment by qzed** [comment]
It's not pressure related. I thought that initially, but it turns out that this is something in the kernel.

**Comment by m1nicrusher** [comment]
> It's not pressure related. I thought that initially, but it turns out that this is something in the kernel.

Oh yes, that issue initially talks about the touchpad but soon the topic was shifted to the random freeze of both keyboard and the touchpad. Just ignore the first few threads.

**Comment by RichardODonoghue** [comment]
I can validate I am also having this issue since upgrading to Ubuntu 23.04 with kernel 6.2. Unsure if the distro upgrade was related as I upgraded both at the same time.

**Comment by qzed** [comment]
The latest releases (v6.2.14 / v6.2.13-2) contain a fix for #1026. Can you give those a test and see if the problem remains?

**Comment by qzed** [comment]
Closing as duplicate of #1026, waiting for feedback there.
